### DISCUSSION TITLE
Using with HA

### DISCUSSION THREAD

**Discussion by UV-PWRD** [jblance/mpp-solar#167] [Q&A]
I've got a Pi4, very new to this, and have Home Assistant running on it natively, as I understand it, it's using HassOS

I've got a JK/Heltec BMS and an MPP 7248, I'm trying to get data out of both for some automation. Ideally I'd like to talk to the BMS by Bluetooth. I do have an RS485 to USB converter for the inverter but it only pulled data into the windows app (watch power) once and then refused to do it again so that adapter might be suspect. 

How do I go about running this python code within HA?

**Comment by jblance** [comment]
It not setup to run 'in' HA - I use HA, but with a different Pi running mppsolar and integrating with HA using MQTT autodiscovery
From memory there is a HA integration for mpp solar (and maybe other) inverters (I've not used it so no idea if it will work for you or not)
https://community.home-assistant.io/t/programmatically-read-data-from-your-solar-inverter-voltronic-axpert-mppsolar-pip-voltacon-effekta-etc-and-interface-with-home-assistant-via-mqtt-works-with-rs232-usb/119053

**Reply by UV-PWRD** [reply]
Thank you, I don't think that will talk to the JK though and that's what I'm really after so I can monitor battery capacity properly. 

The inverter just uses voltage which doesn't tell the full story with Lifepo4 

Was hoping to avoid needing a second Pi.

**Comment by jblance** [comment]
Ahh - I see

Well options as I see it:
1 another Pi as mentioned
2 a ESP to connect to inverter and mqtt to HA on the pi (someone has done a
ESPHome integration for JK)
3 run HA as container or venv on native Pi (this will allow you to run the
python stuff but loses a lot of the convenience functions of HA
4 write a HA integration for mppsolar
5 wait until jblance sorts a HA integration :-)

I personally do 1 and 3, 1 or 2 is probably the easiest

**Reply by michas79de** [reply]
Good list! 👍

I'm using option 3, but I'm not very happy with it.
Any progress with Option 5? ;-)

Option 1 includes buying another expensive Pi, which I don't want to.
Or do you think an old Pi 1 would do for your mpp-solar?

**Reply by m4rcSA** [reply]
Yes, and old Pi 1 does.
Im currently using an old bananapipro with armbian and docker.

**Reply by FrankieHAi** [reply]
How's option 5 looking?

**Reply by jblance** [reply]
Honestly I cant see a way to do it in a sensible way - the ESPHome stuff seems to be a better solution (only requires a ESP instead of a Pi and has 2 way comms)
I was thinking of option 5 being a more complete solution, eg inverter + BMS + shunt into a full something, but honestly havent been able to envisage what that would look like
